### PR TITLE
MQE-2016: Add warning for test materials that violate naming convention

### PR DESCRIPTION
**Pull request by tomreece**
Build:
https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/30465/
Note: This build has failures around video tests, but this is also a problem in mainline right now too.

### CODE REVIEW

File: src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php [hunk 1]
```diff
@@ -297,4 +338,32 @@ private function extendDataObject($dataObject)
...
+     *
+     * @throws TestFrameworkException
+     * @return void
+     */
+    private function logViolationSummary()
+    {
+        $nameViolationCount = count($this->nameViolations);
+        if ($nameViolationCount > 0) {
+            LoggingUtil::getInstance()->getLogger(self::class)->notification(
+                "{$nameViolationCount} data entity name violations detected. Data entity names must be"
```
Comment (soumyau) [inline src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php]: To direct users to detailed log, how how about appending "See mftf.log for details." to notice messages

File: src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php [hunk 2]
```diff
@@ -297,4 +338,32 @@ private function extendDataObject($dataObject)
...
         return $dataObject;
     }
+
+    /**
+     * Output to the log a summary of data entity name and data entity key violations.
+     *
+     * @throws TestFrameworkException
+     * @return void
+     */
+    private function logViolationSummary()
```
[the remark this answers is not in the document]
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php]: fixed

File: src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php (hunk 1 above)
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/DataObjectHandler.php]: fixed

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php [hunk 3]
```diff
@@ -50,4 +68,246 @@ public static function validateName($name, $type)
...
+
+    /**
+     * Validates data entity names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityName($str, $filename = null)
```
Comment (soumyau) [inline src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: can this be repurposed as below for all pascal case validations? There won't be a need to separate function for each entity.
```
validateEntityInPascalCase($str, $type, $filename = null)
{
..
$message = "The ($type) {$str} should be PascalCase with an uppercase first letter.";
..
}
```

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php [hunk 4]
```diff
@@ -50,4 +68,246 @@ public static function validateName($name, $type)
...
+
+    /**
+     * Validates data entity key names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityKey($str, $filename = null)
```
Comment (soumyau) [inline src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: can this be repurposed as below for all camel case validations?
```
validateEntityInCamelCase($str, $type, $filename = null)
{
..
$message = "The ($type) {$str} should be PascalCase with an uppercase first letter.";
..
}
```

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php [hunk 5]
```diff
@@ -50,4 +68,246 @@ public static function validateName($name, $type)
...
+
+    /**
+     * Validates page names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validatePageName($str, $filename = null)
```
Comment (soumyau) [inline src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: can this be repurposed for all prefix, suffix validations?
```
public function validateEntityNameAffixes($str, $type, $filename = null)
{
$isPrefixAdmin = substr_compare($str, self::PREFIX_ADMIN, 0, strlen(self::PREFIX_ADMIN)) === 0;
$isPrefixAdmin = substr_compare($str, self::PREFIX_STOREFRONT, 0, strlen(self::PREFIX_STOREFRONT)) === 0;
$isSuffix = substr_compare($str, $type, 0, strlen($type)) === 0;
..
 $message = "The {$type} name {$str} should follow the pattern {Admin or Storefront}{Description}Page.";
..
}
```

File: src/Magento/FunctionalTestingFramework/Test/Handlers/TestObjectHandler.php [hunk 6]
```diff
@@ -153,7 +156,7 @@ private function initTestData()
             }
         }
         $exceptionCollector->throwException();
-        
+        $testNameValidator->summarize("test name");
```
Comment (soumyau) [inline src/Magento/FunctionalTestingFramework/Test/Handlers/TestObjectHandler.php]: maybe add all types as constants eg: TYPE_TEST_NAME

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php (hunk 3 above)
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: fixed

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php (hunk 4 above)
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: fixed

File: src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php (hunk 5 above)
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php]: fixed

File: src/Magento/FunctionalTestingFramework/Test/Handlers/TestObjectHandler.php (hunk 6 above)
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Test/Handlers/TestObjectHandler.php]: fixed

**Pull request merged**